packages/frontend/src/components/NavBar/ProjectSwitcher.tsx

```diff
@@ -240,13 +245,49 @@ const ProjectSwitcher = () => {
         return projects.find((p) => p.projectUuid === activeProjectUuid);
     }, [activeProjectUuid, projects]);
 
+    // user has permission to create preview project on an organization level
+    const orgRoleCanCreatePreviews = useMemo(() => {
+        return user.data?.ability.can(
+            'create',
+            subject('Project', {
+                organizationUuid: user.data.organizationUuid,
+                type: ProjectType.PREVIEW,
+            }),
+        );
+    }, [user.data]);
+
     const inactiveProjects = useMemo(() => {
         if (!activeProjectUuid || !projects) return [];
-        return projects.filter((p) => p.projectUuid !== activeProjectUuid);
-    }, [activeProjectUuid, projects]);
+        return projects
+            .filter((p) => p.projectUuid !== activeProjectUuid)
+            .filter((project) => {
+                switch (project.type) {
+                    case ProjectType.DEFAULT:
+                        return true;
+                    // check if user has permission to create preview project on an organization level (developer, admin)
+                    // or check if user has permission to create preview project on a project level
+                    // - they should have permission (developer, admin) to the upstream project
```

Comment:
But we don't have that information (the upstream project) here. 
We are assuming the user role was copied from the upstream project to the preview. As in, if they were developers in the upstream project they are developers in the preview so they are allowed to create preview project from the preview project.

Reply:
> they are allowed to create preview project from the preview project.
goo dpoint, we do not handle this yet. I'll make the change!